### DISCUSSION TITLE
Same Port like Webserver

### DISCUSSION THREAD

**Discussion by Peter-Stern** [mockoon/mockoon#404] [Help] [answered]
Hi
I am a newbie and I wanted to test my Frontend without changing code or url. Therefore I am happy to found mockoon. It is a great tool. My Webserver is runnning on localhost:81 and I access my backend with a post-fetch to this url:"./api" (localhost:81/api). To reach the mockoon I have to set the  url to "localhost:3000/api". If I change the port to 81 in mockoon to 81 (same port like my webserver) it's not working. 
Can anyone give me a hint, please?
Thanks in advance
Peter

**Comment by 255kb** [comment, marked as answer]
Hi @PeterSt78, the ports on your machine can only be used by one app at a time. 
The easiest thing to do is to [use Mockoon as a proxy](https://mockoon.com/tutorials/partial-mocking-proxy/). Like this, you will be able to mock new/missing endpoints in Mockoon, and redirect to your existing web server. Still, you will have to either run Mockoon on port 81, and change your web server port to something else, or run Mockoon on another port (e.g. 3000), and proxy to your web server on port 81. But then, you would have to change the port in your frontend.

**Reply by Peter-Stern** [reply]
Hi @255kb 
Thanks. 
In the fetch url I will normally only have "./cgi-api/". So there is not an defiition of port in the javascript, because it is an internally call. if there is no possibility for a "sniffer", I have to make a url-file to switch between test-environment and product-environment.
